src/test/base58_tests.cpp

```diff
@@ -218,14 +247,39 @@ BOOST_AUTO_TEST_CASE(base58_keys_valid_gen)
             else if(exp_addrType == "none")
             {
                 dest = CNoDestination();
-            }
-            else
-            {
+            } else if (exp_addrType == "p2wpkh") {
+                WitnessV0KeyHash id;
+                memcpy(id.begin(), exp_payload.data(), 20);
+                dest = id;
+            } else if (exp_addrType == "p2wsh") {
+                WitnessV0ScriptHash id;
+                memcpy(id.begin(), exp_payload.data(), 32);
+                dest = id;
+            } else if (exp_addrType.size() == 5 && exp_addrType.substr(0, 4) == "p2w?") {
+                WitnessUnknown unk;
+                memcpy(unk.program, exp_payload.data(), exp_payload.size());
+                unk.length = exp_payload.size();
+                unk.version = exp_addrType[4] - 'a';
```

Comment:
gross.  Can these tests be changed to work with whole script pubkeys rather than tweaking the addrType?

Reply:
> gross

I agree. 

> Can these tests be changed to work with whole script pubkeys rather than tweaking the addrType?

Done. I've added an extra commit to first refactor the tests into testing scriptPubKeys.